### PR TITLE
CIV-8308 Judge's Screen Order Details Flight Delay Section

### PR DESCRIPTION
**Pull request by Omaira-Melo-Hmcts**
### JIRA link (if applicable) ###

[Judge's Screen: Order Details - Flight Delay Section (Small Track SDO)](https://tools.hmcts.net/jira/browse/CIV-8308)

### Change description ###
When a judge is making a Small Track SDO, upon selecting the Flight Delay from the option list, when the judge go to the Order Details screen, the section should be available for the Judge to make the order for Flight Delay.
image
The Flight Delay paragraph, with a Show/Remove checkbox, will be visible on the Judge's Order Details screen.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/model/sdo/SmallClaimsFlightDelay.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import java.util.List;
+
+@Data
+@Builder(toBuilder = true)
+@NoArgsConstructor
+@AllArgsConstructor
+public class SmallClaimsFlightDelay {
+
+    private String input1;
```
Comment (Azam-Hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/model/sdo/SmallClaimsFlightDelay.java:17]: Please put specific names for text boxes

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java [hunk 2]
```diff
@@ -604,6 +605,23 @@ private CallbackResponse prePopulateOrderDetailsPages(CallbackParams callbackPar
 
         updatedData.smallClaimsWitnessStatement(tempSmallClaimsWitnessStatement).build();
 
+        SmallClaimsFlightDelay tempSmallClaimsFlightDelay = SmallClaimsFlightDelay.builder()
```
Comment (Azam-Hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java:608]: Add changes behind the featureToggle.

File: src/main/java/uk/gov/hmcts/reform/civil/model/sdo/SmallClaimsFlightDelay.java (hunk 1 above)
Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/model/sdo/SmallClaimsFlightDelay.java]: Azam, we have renamed to be more meaningful as you suggested.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java (hunk 2 above)
Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java]: Have added a condition with R2 toggle as suggested.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java [hunk 3]
```diff
@@ -604,6 +605,25 @@ private CallbackResponse prePopulateOrderDetailsPages(CallbackParams callbackPar
 
         updatedData.smallClaimsWitnessStatement(tempSmallClaimsWitnessStatement).build();
 
+        if (featureToggleService.isSdoR2Enabled()) {
+            SmallClaimsFlightDelay tempSmallClaimsFlightDelay = SmallClaimsFlightDelay.builder()
+                .smallClaimsFlightDelayToggle(checkList)
+                .relatedClaimsInput("In the event that the Claimant(s) or Defendant(s) are aware if other \n"
+                            + "claims relating to the same flight they must notify the court  \n"
+                            + "where the claim is being managed within 14 days of receipt of \n"
+                            + "this Order  providing all relevant details of those claims including \n"
```
Comment (hmcts-version1-ignacio) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java:614]: There are two spaces between "Order" and "providing"

Comment (Omaira-Melo-Hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java]: Two space removed

**Review by ShwetaTandel-hmcts — APPROVED** [review]

**Review by hmcts-version1-ignacio — APPROVED** [review]

**Review by Azam-Hmcts — APPROVED** [review]

**Pull request merged**